xapian-bindings/android/CROSS_COMPILE

```diff
@@ -0,0 +1,121 @@
+use arm-linux-androideabi-4.9. For more info refer https://developer.android.com/ndk/guides/standalone_toolchain.html
+
+<target-android-api> is the target Android platform/API level. eg android-16
+
+<path_to_ndk_standalone_toolchain> is the path where your standalone toolchain will be installed.
+
+2)
+As Xapian makes use of built-in UUID APIs and ndk does not have UUID APIs, we need to explicitly cross compile that as well. So download libuuid, inside the e2fsprogs package by doing following:
+
+git clone git://git.kernel.org/pub/scm/fs/ext2/e2fsprogs.git
+cd e2fsprogs/
+
+Now configure and install the library by doing:
+export TOOLCHAIN=<path_to_ndk_standalone_toolchain>
+export PATH=$PATH:$TOOLCHAIN/bin
+
+./configure --host=<host> --prefix="$TOOLCHAIN/sysroot/usr" CFLAGS="-fPIC"
+cd lib/uuid
+make
+make install
+
+<host> is the target architecture. eg. arm-linux-androideabi
+<target-arch> is the same as in step 1)
+
+3)
+For testing purpose you will need adb installed.
+
+Cross compiling Xapian
+======================
+Now cross compile xapian for android as follows:
+
+export TOOLCHAIN=<path_to_ndk_standalone_toolchain>
+export PATH=$PATH:$TOOLCHAIN/bin
+export OUTPUT_DIR=<path_to_xapian>/xapian-bindings/android/built
+cd <path_to_xapian>/xapian-core
+./configure --prefix=$OUTPUT_DIR --host=<host>  LDFLAGS="-fPIE -pie" --disable-shared --with-pic 
+make
+make install
+
+The above configure will build xapian statically with position independent code.
+
+Generating the bindings for android
+===================================
+Now generate the bindings for android as follows:
+
+cd <path_to_xapian>/xapian-bindings/android
+mkdir -p org/xapian
+swig -Werror -I. -I../../xapian-core/include -c++ -java -module Xapian -package org.xapian -outdir org/xapian -o xapian_wrap.cc android.i
+
+This should generate xapian_wrap.cc,xapian_wrap.h and .java files in org/xapian. Now add the following line in org/xapian/XapianJNI.java at start of static block at line no 1114:
+
+System.loadLibrary("xapian_jni");
+
+Save the file. Now generate the libxapian_jni.so by doing the following:
+
+export CXX=$TOOLCHAIN/bin/<host>-g++
+$CXX -c xapian_wrap.cc -I../../xapian-core/include
+$CXX -shared -o built/lib/libxapian_jni.so -Wl,-whole-archive libxapian-x.x.a  -Wl,-no-whole-archive xapian_wrap.o -lz -luuid -lc
+
+This will generate libxapian_jni.so in built/lib/libxapian_jni.so. Now compile all the .java class using :
+
+javac -source 1.7 -target 1.7 -bootclasspath <path_to_jvm>/<your_jdk>/jre/lib/rt.jar org/xapian/*.java
+
+<path_to_jvm> is the jvm installation path eg.usr/lib on Ubuntu
+The 'source' and 'target' flags are required for android compactibility if you have java 1.8 or above. 
+
+
+Now create a proper file structure for .aar for android by:
+
+mkdir -p aar/jni/<proper_architecture> aar/res
+jar -cf aar/classes.jar org/xapian/*.class
+cp built/lib/libxapian_jni.so aar/jni/<proper_architecture>/
```

Comment:
Also, http://tools.android.com/tech-docs/new-build-system/aar-format implies there are some mandatory files we aren't creating here - ` AndroidManifest.xml` and`R.txt`.  Should we be generating those somehow?